### PR TITLE
Add option to specify shared libs to load in llhd-sim

### PR DESCRIPTION
**Pull request by mikeurbach**

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -14,14 +14,11 @@ set(CIRCT_TEST_DEPENDS
   circt-translate
   handshake-runner
   firtool
+  llhd-sim
   mlir-opt
   mlir-cpu-runner
   )
 
-if(NOT WIN32)
```
Comment (teqdruid) [inline test/CMakeLists.txt:21]: Do the tests now work on Windows or does it just not cause the Windows build to fail?

Comment (mikeurbach) [reply test/CMakeLists.txt:21]: I'm not sure yet :) I want to trigger the test manually and see what happens.

**Comment by lattner** [comment]
Code looks nice to me

**Comment by mikeurbach** [comment]
I manually triggered the following:
Nightly integration tests: https://github.com/llvm/circt/actions/runs/666647249
Windows tests: https://github.com/llvm/circt/actions/runs/666647766

Both passed. Does this resolve #708 and #375 ?

**Comment by teqdruid** [comment]
> Both passed. Does this resolve #708 and #375 ?

I think so, yes. Boo ya! Thanks! 🙇‍♂️

**Review by teqdruid — COMMENTED** [review]
This looks good to me but Fabian should really be the one approving. It seems like an acceptable solution to what's turned out to be an ugly problem in terms of compatibility (esp. on Windows).

File: tools/llhd-sim/llhd-sim.cpp [hunk 2]
```diff
@@ -102,6 +102,10 @@ static cl::opt<TraceFormat> traceMode(
             "instance and signals not having the default name '(sig)?[0-9]*'"),
         clEnumValN(noTrace, "no-trace", "Don't dump a signal trace")));
 
+static cl::list<std::string>
+    sharedLibs("shared-libs", cl::desc("Libraries to link dynamically"),
+               cl::ZeroOrMore, cl::MiscFlags::CommaSeparated);
```
Comment (mikeurbach) [inline tools/llhd-sim/llhd-sim.cpp:107]: FWIW this new flag is totally optional, and should only be required in the situations that didn't work previously.

**Review by fabianschuiki — APPROVED** [review]
Cool, looks great to me! Thanks for fixing this :+1: :tada: . I'm still surprised this caused so much hassle; feels to me like the JITed code should have just been able to pick up some of these symbols from the binary itself. But it's super nice to have this work again!

**Comment by youngar** [comment]
Nice work!  Thanks for figuring this out

**Comment by mikeurbach** [comment]
> should have just been able to pick up some of these symbols from the binary itself

Agreed. I didn't do any spelunking on the code, but it seems like the flag to add shared libs was added upstream for this exact reason. There was a related comment in this code that specifically mentioned GDB and symbols.

**Pull request merged**